features/support/steps.js

```diff
@@ -25,8 +25,24 @@ Before(function hook() {
 });
 
 After(async function hook() {
-  fs.remove(this.dir);
-  await util.promisify(kill)(this.proc.pid);
+  // kill all running processes
+  const pids = [];
+  try {
+    pids.push(...await pidtree(this.proc.pid));
+  } catch (error) {
+    // the process doesn't exist anymore
+  }
+  pids.forEach((pid) => {
+    try {
+      process.kill(pid, 'SIGKILL');
+    } catch (error) {
+      // re-throw except in case it is 'ESRCH' (process cannot be found)
+      if (error.code !== 'ESRCH') throw error;
+    }
+  });
+
+  // remove the temporary directory
+  return fs.remove(this.dir);
 });
```

Comment:
`fs.remove()` returns a promise, which was previously ignored... the other issues were that:

- tree-kill didn't kill the processes very well
- the directory was removed before the process got killed (caused problems with Ruby hooks)